Fix for syntax error when compiling with GCC 7.3.0.
Taken from https://develop.openfoam.com/Development/ThirdParty-plus/commit/1a1c62442250f1cd028ce8f90bb1ca82c33ac404

Henrik R. Nagel, 20190912
diff -ru ParaView-v5.5.2.orig/VTK/ThirdParty/hdf5/vtkhdf5/src/H5detect.c ParaView-v5.5.2/VTK/ThirdParty/hdf5/vtkhdf5/src/H5detect.c
--- ParaView-v5.5.2.orig/VTK/ThirdParty/hdf5/vtkhdf5/src/H5detect.c	2018-06-15 21:09:45.000000000 +0200
+++ ParaView-v5.5.2/VTK/ThirdParty/hdf5/vtkhdf5/src/H5detect.c	2019-08-19 12:41:56.000000000 +0200
@@ -1360,8 +1360,8 @@
  *
  *-------------------------------------------------------------------------
  */
-static void
-detect_C89_integers(void) HDF_NO_UBSAN
+static void HDF_NO_UBSAN
+detect_C89_integers(void)
 {
     DETECT_BYTE(signed char,	  SCHAR,        d_g[nd_g]); nd_g++;
     DETECT_BYTE(unsigned char,	  UCHAR,        d_g[nd_g]); nd_g++;
@@ -1388,8 +1388,8 @@
  *
  *-------------------------------------------------------------------------
  */
-static void
-detect_C89_floats(void) HDF_NO_UBSAN
+static void HDF_NO_UBSAN
+detect_C89_floats(void)
 {
     DETECT_F(float,     FLOAT,      d_g[nd_g]); nd_g++;
     DETECT_F(double,    DOUBLE,     d_g[nd_g]); nd_g++;
@@ -1410,8 +1410,8 @@
  *
  *-------------------------------------------------------------------------
  */
-static void
-detect_C99_integers8(void) HDF_NO_UBSAN
+static void HDF_NO_UBSAN
+detect_C99_integers8(void)
 {
 #if H5_SIZEOF_INT8_T>0
   #if H5_SIZEOF_INT8_T==1
@@ -1472,8 +1472,8 @@
  *
  *-------------------------------------------------------------------------
  */
-static void
-detect_C99_integers16(void) HDF_NO_UBSAN
+static void HDF_NO_UBSAN
+detect_C99_integers16(void)
 {
 #if H5_SIZEOF_INT16_T>0
     DETECT_I(int16_t, 		  INT16,        d_g[nd_g]); nd_g++;
@@ -1510,8 +1510,8 @@
  *
  *-------------------------------------------------------------------------
  */
-static void
-detect_C99_integers32(void) HDF_NO_UBSAN
+static void HDF_NO_UBSAN
+detect_C99_integers32(void)
 {
 #if H5_SIZEOF_INT32_T>0
     DETECT_I(int32_t, 		  INT32,        d_g[nd_g]); nd_g++;
@@ -1548,8 +1548,8 @@
  *
  *-------------------------------------------------------------------------
  */
-static void
-detect_C99_integers64(void) HDF_NO_UBSAN
+static void HDF_NO_UBSAN
+detect_C99_integers64(void)
 {
 #if H5_SIZEOF_INT64_T>0
     DETECT_I(int64_t, 		  INT64,        d_g[nd_g]); nd_g++;
@@ -1599,8 +1599,8 @@
  *
  *-------------------------------------------------------------------------
  */
-static void
-detect_C99_integers(void) HDF_NO_UBSAN
+static void HDF_NO_UBSAN
+detect_C99_integers(void)
 {
     /* break it down to more subroutines so that each module subroutine */
     /* is smaller and takes less time to compile with optimization on.  */
@@ -1625,8 +1625,8 @@
  *
  *-------------------------------------------------------------------------
  */
-static void
-detect_C99_floats(void) HDF_NO_UBSAN
+static void HDF_NO_UBSAN
+detect_C99_floats(void)
 {
 #if H5_SIZEOF_DOUBLE == H5_SIZEOF_LONG_DOUBLE
     /*
@@ -1656,8 +1656,8 @@
  *
  *-------------------------------------------------------------------------
  */
-static void
-detect_alignments(void) HDF_NO_UBSAN
+static void HDF_NO_UBSAN
+detect_alignments(void)
 {
     /* Detect structure alignment for pointers, hvl_t, hobj_ref_t, hdset_reg_ref_t */
     DETECT_M(void *,              POINTER,      m_g[na_g]); na_g++;
@@ -1744,8 +1744,8 @@
  *
  *-------------------------------------------------------------------------
  */
-int
-main(void) HDF_NO_UBSAN
+int HDF_NO_UBSAN
+main(void)
 {
 
 #if defined(H5_HAVE_SETSYSINFO) && defined(SSI_NVPAIRS)
